ggml-sycl.cpp

```diff
@@ -17463,6 +17486,17 @@ GGML_API GGML_CALL int ggml_backend_sycl_get_device_index(int device_id) {
     return g_sycl_gpu_mgr->get_index(device_id);
 }
 
+GGML_API GGML_CALL void ggml_backend_sycl_set_single_device(int main_gpu_id) {
+    GGML_ASSERT(main_gpu_id<g_all_sycl_device_count);
+    printf("ggml_backend_sycl_set_single_device: use single device: %d\n", main_gpu_id);
+    if (!g_sycl_gpu_mgr) {
+        delete g_sycl_gpu_mgr;
+    }
```

Comment:
Should this be `if (g_sycl_gpu_mgr)` instead?

Reply:
Yes, it's mistake. :)
I will fix it as soon.

Reply:
In this case, recreate a sycl_gpu_mgr() with new GPU list.
before create new one, delete existed instance.